src/servers/trigger.ts

```diff
@@ -130,7 +130,7 @@ export async function onchainBlockEventhandler({ height }) {
 export const onInvoiceUpdate = async (invoice: GetInvoiceResult) => {
   logger.debug({ invoice }, "onInvoiceUpdate")
 
-  if (!invoice.is_confirmed) {
+  if (!invoice.is_held) {
```

Comment:
I just did some testing and as far as I can tell `is_held` carries the following values in 3 different states:
- `undefined`: before the invoice is paid
- `true`: when invoice is paid but not settled
- `undefined`: when invoice is paid & settled

If this is right, it means that `!invoice.is_held` is likely always `true` since (I assume) we do immediate settlement on receipt from the application and there's no meaningful "hold" state.

Reply:
there is no action needed if we received an invoice update where ` invoice.is_confirmed === true`. the ledger update had already happened.